sklearn/svm/_classes.py

```diff
@@ -271,6 +293,22 @@ def fit(self, X, y, sample_weight=None):
         check_classification_targets(y)
         self.classes_ = np.unique(y)
 
+        if self.dual == "auto":
+            self._dual = _choose_dual_automatically(
```

Comment:
`self._dual` seems to be only used in `fit`, which means we don't have to attach it to the instance. Could simply be a local variable, same with the other class.